src/avfoundation.cs

```diff
@@ -9387,19 +9651,20 @@ interface AVCaptureFileOutput {
 		[Export ("stopRecording")]
 		void StopRecording ();
 
-		[NoiOS]
+		[NoiOS, MacCatalyst (15,0)]
 		[Export ("pauseRecording")]
 		void PauseRecording ();
 
-		[NoiOS]
+		[NoiOS, MacCatalyst (15,0)]
 		[Export ("resumeRecording")]
 		void ResumeRecording ();
 
-#if MONOMAC
-		[Unavailable (PlatformName.MacCatalyst)]
+#if MONOMAC || __MACCATALYST__
+		[NoMacCatalyst]
```

Comment:
You're adding `|| __MACCATALYST__` and then also adding `[NoMacCatalyst]`.

In this case it's better to move the `#endif` further up (to not include the RecordingPaused property), and then add  `[NoiOS]` on that property.

As long as reasonably possible, we should strive towards removing/reducing `#if <platform>` code, not make it more complex.